src/util/file_log.rs

```diff
@@ -120,13 +117,22 @@ impl RotatingFileLogger {
     }
 }
 
-impl LogWriter for RotatingFileLogger {
-    fn write(&self, args: Arguments) {
+impl Write for RotatingFileLogger {
+    fn write(&mut self, bytes: &[u8]) -> Result<usize, ::std::io::Error> {
```

Comment:
Looks like `FullFormat` will call `write` [multiple times](https://docs.rs/slog-term/2.0.0-alpha1/src/slog_term/lib.rs.html#253) for one line of log. Is `write` trait sufficient enough?

Reply:
Hm, I think it is sufficient but not ideal now that you point this out. Especially when we're talking about something that is generally making system calls every write.

Perhaps we should be wrapping it with a [`BufWriter`](https://doc.rust-lang.org/std/io/struct.BufWriter.html)?

Reply:
It still calls `should_rollover` frequently, which issues system call too. Now that it will flush when a row is finished, maybe the check can just be done in `flush` method. Besides, the logger is only used in a thread, so the lock to inner core seems redundant.

Reply:
You're totally right! This is some great feedback, I'll refactor this. :+1: